cl/search/api_utils.py

```diff
@@ -36,6 +36,23 @@
 logger = logging.getLogger(__name__)
 
 
+class ResultObject:
+    def __init__(self, initial=None):
+        self.__dict__["_data"] = initial or {}
+
+    def __getattr__(self, key):
+        return self._data.get(key, None)
+
+    def to_dict(self):
+        return self._data
+
+
+class ESResultObject(ResultObject):
+
+    def __getattr__(self, key):
+        return getattr(self._data, key, None)
```

Comment:
Instead of this custom class, let's use the built-in `defaultdict` class. This will simplify the code and provides automatic default values for missing attributes or keys when accessed.